packages/web-components/src/menu-item/menu-item.ts

```diff
@@ -64,7 +99,17 @@ export class MenuItem extends FASTElement {
    */
   @attr({ mode: 'boolean' })
   public checked: boolean = false;
-  protected checkedChanged(oldValue: boolean, newValue: boolean): void {
+
+  /**
+   * Handles changes to checked attribute custom states and element internals
+   * @param prev - the previous state
+   * @param next - the next state
+   */
+  protected checkedChanged(prev: boolean, next: boolean): void {
+    const checkableMenuItem = this.role !== MenuItemRole.menuitem;
+    this.elementInternals.ariaChecked = checkableMenuItem ? `${!!next}` : null;
```

Comment:
Do we need to force boolean state here?

Reply:
If it's a `menuitemcheckbox` or `menuitemradio` and null is passed we want to force a boolean state of `false`. In the event it's not a checkable instance we want it to always be null as it's not able to be checked.